src/test/java/org/elasticsearch/index/query/RangeQueryBuilderTest.java

```diff
@@ -0,0 +1,163 @@
+        }
+        return query;
+    }
+
+    @Override
+    protected void assertLuceneQuery(RangeQueryBuilder queryBuilder, Query query, QueryParseContext context) throws IOException {
+        assertThat(query.getBoost(), is(queryBuilder.boost()));
+        String fieldname = queryBuilder.fieldname();
+        if (!fieldname.equals(DATE_FIELD_NAME) && !fieldname.equals(INT_FIELD_NAME)) {
+            assertThat(query, instanceOf(TermRangeQuery.class));
+            TermRangeQuery termRangeQuery = (TermRangeQuery) query;
+            assertThat(termRangeQuery.includesLower(), is(queryBuilder.includeLower()));
+            assertThat(termRangeQuery.includesUpper(), is(queryBuilder.includeUpper()));
+            assertThat(termRangeQuery.getLowerTerm(), is(BytesRefs.toBytesRef(queryBuilder.from())));
+            assertThat(termRangeQuery.getUpperTerm(), is(BytesRefs.toBytesRef(queryBuilder.to())));
+            assertThat(termRangeQuery.getField(), is(fieldname));
+        } else if (fieldname.equals(DATE_FIELD_NAME)) {
+            assertThat(query, instanceOf(LateParsingQuery.class));
+            Long min = expectedDateLong(queryBuilder.from(), queryBuilder, context);
+            Long max = expectedDateLong(queryBuilder.to(), queryBuilder, context);
+            Query expectedQuery = NumericRangeQuery.newLongRange(DATE_FIELD_NAME, min, max, queryBuilder.includeLower(), queryBuilder.includeUpper());
+            assertEquals(query.rewrite(null), expectedQuery.rewrite(null));
+        } else {
+            assertThat(query, instanceOf(NumericRangeQuery.class));
+            Query expectedQuery = NumericRangeQuery.newIntRange(INT_FIELD_NAME, (Integer) queryBuilder.from(), (Integer) queryBuilder.to(), queryBuilder.includeLower(), queryBuilder.includeUpper());
+            expectedQuery.setBoost(testQuery.boost());
+            assertEquals(query, expectedQuery);
+        }
+        if (queryBuilder.queryName() != null) {
+            Query namedQuery = context.copyNamedFilters().get(queryBuilder.queryName());
+            assertThat(namedQuery, equalTo(query));
+        }
+    }
+
+    @Test
+    public void testValidate() {
+        RangeQueryBuilder rangeQueryBuilder = new RangeQueryBuilder("");
+        assertThat(rangeQueryBuilder.validate().validationErrors().size(), is(1));
+
+        rangeQueryBuilder = new RangeQueryBuilder("okay").timeZone("UTC");
+        assertNull(rangeQueryBuilder.validate());
+
+        rangeQueryBuilder.timeZone("blab");
+        assertThat(rangeQueryBuilder.validate().validationErrors().size(), is(1));
+
+        rangeQueryBuilder.timeZone("UTC").format("basicDate");
+        assertNull(rangeQueryBuilder.validate());
+
+        rangeQueryBuilder.timeZone("UTC").format("broken_xx");
+        assertThat(rangeQueryBuilder.validate().validationErrors().size(), is(1));
+
+        rangeQueryBuilder.timeZone("xXx").format("broken_xx");
+        assertThat(rangeQueryBuilder.validate().validationErrors().size(), is(2));
```

Comment:
maybe we can also test what happens when we receive the timezone but the field is a number, and the other case of "validation" that we perform based on the mapping? Having the mappings around is great, we can test much more now!